tests/test_core.py

```diff
@@ -8,15 +10,52 @@ class test_Interpolator(unittest.TestCase):
 
     def test_init(self):
         """Test creation from data."""
-        raise NotImplementedError()
+
+        #test construction with tuple input
+        f = relentless.core.Interpolator(x=(-1,0,1), y=(-2,0,2))
+        self.assertEqual(f._domain, (-1,1))
+        self.assertIsInstance(f._spline, Akima1DInterpolator)
+
+        #test construction with list input
+        f = relentless.core.Interpolator(x=[-1,0,1], y=[-2,0,2])
+        self.assertEqual(f._domain, (-1,1))
+        self.assertIsInstance(f._spline, Akima1DInterpolator)
+
+        #test construction with numpy array input
+        f = relentless.core.Interpolator(x=np.array([-1,0,1]), y=np.array([-2,0,2]))
+        self.assertEqual(f._domain, (-1,1))
+        self.assertIsInstance(f._spline, Akima1DInterpolator)
+
+        #test construction with non-increasing domain
+        with self.assertRaises(ValueError):
+            f = relentless.core.Interpolator(x=(1,0,-1), y=(2,0,-2))
 
     def test_call(self):
         """Test calls, both scalar and array."""
-        raise NotImplementedError()
+        f = relentless.core.Interpolator(x=(-1,0,1), y=(-2,0,2))
+
+        #test scalar call
+        self.assertAlmostEqual(f(-0.5), -1.0)
+        self.assertAlmostEqual(f(0.5), 1.0)
+
+        #test array call
+        np.testing.assert_almost_equal(f([-0.5,0.5]), [-1.0,1.0])
```

Comment:
NumPy recommends using `assert_allclose` instead of this assertion, so please replace throughout.